app/services/address/submit-select.service.js

```diff
@@ -62,14 +62,17 @@ async function go(sessionId, payload) {
 async function _save(session, address) {
   session.address.uprn = address.uprn
 
-  const premiseseStreetAddress = address.premises + ' ' + address.street_address
+  const premises = address.premises ?? ''
+  const streetAddress = address.street_address ?? ''
+
+  const premisesStreetAddress = premises + ' ' + streetAddress
```

Comment:
How about putting the `trim()` and using interpolation?

```suggestion
  const premisesStreetAddress = `${premises} ${streetAddress}`.trim()
```

We don't usually use concatenation, and this avoids repeating the call to `trim()` below.